### PR TITLE
Concatenate performance improvement.

### PR DESCRIPTION
**Pull request by OzanCansel**
I have seen the repo on your reddit post while discovering the code I have made a little performance improvement to Keyboard.hpp

### CODE REVIEW

File: include/os/keyboard.hpp [hunk 1]
```diff
@@ -234,12 +234,31 @@ struct combination
     /// Append a combination
     combination & operator+=(const combination &combo)
     {
+        keys.reserve( std::size( keys ) + std::size( combo.keys ) );
```
Comment (cskrisz) [inline include/os/keyboard.hpp:237]: This isn't necessary. `insert()` already does max one allocation here.

Comment (OzanCansel) [reply include/os/keyboard.hpp]: You are right, I looked insert implementation, so it does as you said. Removed that line and committed

**Comment by gavrilikhin-d** [comment]
Will this really be faster than copy from this? Because it makes a one-line function much bigger.

**Comment by OzanCansel** [comment]
Yes, it is faster or equal because it guarantees that only one allocation occurs on concatenate. Otherwise, it should resize when capacity is not enough while executing `+= combo;` in `combination operator+(const combination &combo)` so two allocations and one more copy occur.

**Pull request merged**